### ISSUE TITLE
How to restrict the server capabilities by the client?

### ISSUE THREAD

**Issue by przepompownia** [phpactor/language-server-phpactor-extensions#56]
I spent some hours to find how to launch the server with the restricted set of capabilities. Excluding some of them on the nvim built-in client side (e.g. `capabilities.textDocument.definition = nil`) on initialization gave no effect for me. 

I guess (I hope I'm wrong) it's currently not possible. Does then each extension need to be prevented from registering particular handler depending on the client's capabilities?

**Comment by dantleech** [comment]
Yeah, not currently possible to adapt to the client s requirements/restrictions.

Open to ideas, maybe a "isSupported(Client capabilities $capabilities)" on the handler itself?

**Comment by przepompownia** [comment]
Please look on the example approach in #58
